### PR TITLE
Use custom keyfile as gsettings backend

### LINKED ISSUE #73: Isolate desktop apps from the system by default
> I think we should wrap all GTK apps with a `dbus-run-session ...`, this way it should inherit `XDG_CONFIG_HOME`.
> 
> I'll do some experiments.

_Originally posted by @Saviq in https://github.com/MirServer/mir-ci/pull/62#discussion_r1479451635_

### PR DESCRIPTION
**Pull request by hbatagelo**
Fixes #73.

### CODE REVIEW

File: mir-ci/mir_ci/conftest.py [hunk 1]
```diff
@@ -225,3 +225,23 @@ def xdg(request: pytest.FixtureRequest, tmp_path: pathlib.Path) -> Generator:
...
+def env(request: pytest.FixtureRequest) -> Generator:
+    """
+    Set environment variables.
+
+    Use the `env` mark to provide data, e.g.:
+    @pytest.mark.env(ENV_VARIABLE='value')
+    """
+    if request.config.getoption("--deps"):
+        yield
+        return
```
Comment (Saviq) [inline mir-ci/mir_ci/conftest.py:240]: Do we need this? We may even want the env var to affect dep installation, I suppose?

File: mir-ci/mir_ci/conftest.py [hunk 2]
```diff
@@ -225,3 +225,23 @@ def xdg(request: pytest.FixtureRequest, tmp_path: pathlib.Path) -> Generator:
...
+
+    with pytest.MonkeyPatch.context() as m:
+        marks: Iterator[pytest.Mark] = request.node.iter_markers("env")
+        for mark in marks:
+            for var, value in mark.kwargs.items():
```
Comment (Saviq) [inline mir-ci/mir_ci/conftest.py:245]: You should reverse this, as `.iter_markers` will walk up the stack to find the marks, and you need the closest mark to override any previous ones.

**Review by Saviq — CHANGES_REQUESTED** [review]
Tiny tweaks inline

File: mir-ci/mir_ci/conftest.py (hunk 2 above)
Comment (hbatagelo) [reply mir-ci/mir_ci/conftest.py]: Done.

**Review by Saviq — APPROVED** [review]
Cool!

The failure is MirServer/checkbox-mir#9, so nothing to do here.

**Pull request merged**